packages/block-library/src/table-of-contents/index.js

```diff
@@ -19,6 +19,28 @@ export const settings = {
 	icon,
 	edit,
 	save,
+	example: {
```

Comment:
Could this be moved to block.json?

I think the example check above might be unnecessary if we use `innerBlocks` (?)

```json
	"example": {
		"innerBlocks": [
			{
				"name": "core/heading",
				"attributes": {
					"level": 2,
					"content": "Heading"
				}
			},
			{
				"name": "core/heading",
				"attributes": {
					"level": 3,
					"content": "Subheading"
				}
			},
			{
				"name": "core/heading",
				"attributes": {
					"level": 2,
					"content": "Heading"
				}
			},
			{
				"name": "core/heading",
				"attributes": {
					"level": 3,
					"content": "Subheading"
				}
			}
		],
		"attributes": {
			"headings": [
				{
					"content": "Heading",
					"level": 2
				},
				{
					"content": "Subheading",
					"level": 3
				},
				{
					"content": "Heading",
					"level": 2
				},
				{
					"content": "Subheading",
					"level": 3
				}
			]
		}
	},
```


<img width="566" alt="Screenshot 2024-09-23 at 11 45 29 am" src="https://github.com/user-attachments/assets/5a41e450-b76a-4d65-a3de-f74cd08607a9">

Reply:
Yep, that works, thanks 🙇 

I take it that the definition of the `innerBlocks` and them being rendered initially for the example means there are heading blocks in the registry. That then leads the `useObserveHeadings` hook to find those before updating the block's `headings` attribute.

If I remove the `attributes.headings` value, you can see the block example start in the placeholder state then update to match the provided inner blocks.

I've pushed a [commit](https://github.com/WordPress/gutenberg/pull/65549/commits/1613438920cb028d02bb901e79f5cf93fa6e44e6) using the suggested approach.

Reply:
> I take it that the definition of the innerBlocks and them being rendered initially for the example means there are heading blocks in the registry. That then leads the useObserveHeadings hook to find those before updating the block's headings attribute.

That sounds legit. My brain stops functioning when I see something working. 😄